x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -116,9 +117,11 @@ public abstract class ExpressionBuilder extends IdentifierBuilder {
     public static final int MAX_EXPRESSION_DEPTH = 400;
 
     protected final QueryParams params;
+    private final PlanningMetrics metrics;
```

Comment:
Make it protected so subclasses can refer to it instead of having their own copies.
To avoid future disruption in the constructor signature, introduce a basic ParsingContext record (`record ParsingContext(QueryParams params, PlanTelemetry)` and use that instead.

Reply:
Ah, right, this was silly. Thx.